packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -154,85 +160,92 @@ const sharedStyles = StyleSheet.create({
     },
     slider: {
         position: "absolute",
-        top: Spacing.xxxxSmall_2,
-        left: Spacing.xxxxSmall_2,
-        height: `calc(${Spacing.medium_16}px + ${Spacing.xxxSmall_4}px)`,
-        width: `calc(${Spacing.medium_16}px + ${Spacing.xxxSmall_4}px)`,
-        borderRadius: "50%",
-        backgroundColor: Color.white,
-        transition: "transform 0.15s ease-in-out",
+        top: theme.spacing.slider.position,
+        left: theme.spacing.slider.position,
+        height: theme.size.height.medium,
+        width: theme.size.width.medium,
+        borderRadius: theme.border.radius.full,
+        backgroundColor: theme.color.bg.slider.on,
```

Comment:
**question:** What happens if you set this as `backgroundColor: theme.color.bg.slider`? does TS throw an error?

Reply:
discussed offline: it won't throw an error here, but inside the `_generateStyles` function, the `newStyles` object should have a type of `Record<string, CSSProperties>` instead of `Record<string, any>`.